### DISCUSSION TITLE
MTM force publishing

### DISCUSSION THREAD

**Discussion by tangethan79** [jhu-dvrk/sawIntuitiveResearchKit#207] [Q&A] [answered]
Hi I was hoping I could get some ideas on how to smoothly implement force publishing for the dVRK MTMs. 

Currently, I have a Python script which calculates the necessary forces and publishes them through CRTK as servo_cf messages. I found that, when the dVRK console is running, the MTMs behave erratically as both processes fight for control. I see that in the terminal output, the control mode rapidly switches between measured_cp (dvrk console) and servo_cf (my Python script). 

Is there a smooth way of combining these two control modes? Thank you for the help!

**Comment by adeguet1** [comment, marked as answer]
By default the console shouldn't be sending messages to the mtms unless the teleoperation  is turned on.  When teleoperation is on, the logic is to first send commands to the mtms to align with the PSMs.  While teleoperating (follow mode) the logic stops sending commands to the mtms.  Do you need teleoperation on while you're sending forces?   If so, there is a topic to send wrenches on top of the teleoperation logic.  The code will ignore the wrench you send until the arms are in follow mode.  The topic is /MTML_PSM1/following/mtm/body/servo_cf
